modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileFeatureSource.java

```diff
@@ -345,7 +346,17 @@ protected FeatureReader<SimpleFeatureType, SimpleFeature> getReaderInternal(Quer
         // setup the feature readers
         ShapefileSetManager shpManager = getDataStore().shpManager;
         @SuppressWarnings("PMD.CloseResource") // managed as a field of the return value
-        ShapefileReader shapeReader = shpManager.openShapeReader(geometryFactory, goodRecs != null);
+        ShapefileReader shapeReader = null;
+        try {
+            @SuppressWarnings("PMD.CloseResource") // managed as a field of the return value
+            final ShapefileReader sr =
+                    shpManager.openShapeReader(geometryFactory, goodRecs != null);
+            shapeReader = sr;
+        } catch (final FileNotFoundException e) {
+            final String format = "Ignoring missing shp-file and moving on: %s";
+            LOGGER.fine(() -> String.format(format, e.getMessage()));
+        }
```

Comment:
Ouch, relying on exceptions for predictable situations is a bad habit. Is there no way to check if the shapefile is present instead?

Reply:
@aaime 
In case the files are local, the existence of the shp-file can be tested. I implemented this test. If the files are not local, there is no way to avoid catching the FileNotFoundException.